src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -236,22 +237,36 @@ public async Task StopAsync(CancellationToken cancellationToken)
 
         public async Task RestartHostAsync(CancellationToken cancellationToken)
         {
-            if (State == ScriptHostState.Stopping || State == ScriptHostState.Stopped)
+            if (Interlocked.CompareExchange(ref _restarting, 1, 0) != 0)
```

Comment:
Analysis of some of the AppVeyor failues showed the following error in the console output for the test run. I also checked in Kusto and we're seeing this same "Start has already been called" error in production.

```
System.InvalidOperationException : Start has already been called.
   at Microsoft.Azure.WebJobs.JobHost.StartAsync(CancellationToken cancellationToken) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\JobHost.cs : 91
   at Microsoft.Azure.WebJobs.Hosting.JobHostService.StartAsync(CancellationToken cancellationToken) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Hosting\JobHostService.cs : 26
   at async Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Script.WebHost.WebJobsScriptHostService.StartHostAsync(CancellationToken cancellationToken,Int32 attemptCount,JobHostStartupMode startupMode) at C:\azure-webjobs-sdk-script\src\WebJobs.Script.WebHost\WebJobsScriptHostService.cs : 144
```